roles/ceph-osd/tasks/activate_osds.yml

```diff
@@ -20,11 +20,18 @@
     - devices
   changed_when: false
   failed_when: false
+  register: activate_osd_disk
   when:
     not item.0.get("skipped") and
     item.0.get("rc", 0) != 0 and
     not osd_auto_discovery
 
+- name: fail if ceph-disk cannot create an OSD
+  fail:
+    msg: "ceph-disk failed to create an OSD"
+  when: "ceph-disk: Error: ceph osd create failed" in item.get("stderr", "")
```

Comment:
why not ``activate_osd_disk.get('stderr', '')``?  I don't think you need ``with_items`` here do you?

If you want all the lines of stderr I think you can use ``activate_osd_disk.stderr_lines``.